pkg/webhooks/webhookpatch.go

```diff
@@ -101,8 +106,8 @@ func (w *WebhookCertPatcher) runWebhookController(stopChan <-chan struct{}) {
 
 func (w *WebhookCertPatcher) updateWebhookHandler(oldConfig, newConfig *v1.MutatingWebhookConfiguration) {
 	if oldConfig.ResourceVersion != newConfig.ResourceVersion {
-		for i, wh := range newConfig.Webhooks {
-			if strings.HasSuffix(wh.Name, w.webhookName) && !bytes.Equal(newConfig.Webhooks[i].ClientConfig.CABundle, w.caCertPem) {
+		for _, wh := range newConfig.Webhooks {
+			if strings.HasSuffix(wh.Name, w.webhookName) {
```

Comment:
This change seems suspicious, now we patch even when no changes needed?

Reply:
This get triggered  when mutatingwebhookconfiguration added or updated.  For mutatingwebhookconfiguration,  we do not update it after creating.   Since we handle the caBundle update seperately we can remove this useless check.

Reply:
I still don't think its useless.. its good to skip updates when they are not needed, updates are fairly expensive.

Even if we are not triggering updates, they can happen for all sorts of reasons. Why not add a simple 1 line check to avoid a full api server call?

Reply:
add the check back for update case